### PR TITLE
Issue #142: Added yaml option for chaos init

### PR DESCRIPTION
**Pull request by nstjelja**
Issue #142: Added yaml option for chaos init with documentation improvements and assorted unit test

Signed-off-by: Nikola Stjelja <nstjelja@gmail.com>

### CODE REVIEW

File: chaostoolkit/cli.py [hunk 1]
```diff
@@ -385,6 +404,11 @@ def init(ctx: click.Context, discovery_path: str = "./discovery.json",
 with_plugins(iter_entry_points('chaostoolkit.cli_plugins'))(cli)
 
 
+def is_yaml(experiment_path):
```
[the remark this answers is not in the document]
Comment (nstjelja) [reply chaostoolkit/cli.py]: done, is it ok?

File: chaostoolkit/cli.py [hunk 2]
```diff
@@ -253,13 +254,23 @@ def discover(ctx: click.Context, package: str,
...
+              help='Path where to save the experiment (.yaml or .json)',
               show_default=True)
 @click.pass_context
 def init(ctx: click.Context, discovery_path: str = "./discovery.json",
          experiment_path: str = "./experiment.json") -> Experiment:
     """
-    Initialize a new experiment from discovered capabilities.
+    Initialize a new experiment from discovered capabilities:
+
+    ~/chaos init
```
[the remark this answers is not in the document]
Comment (nstjelja) [reply chaostoolkit/cli.py]: done is it ok?

**Comment by nstjelja** [comment]
> LGTM
> 
> I only allow the pyyaml dependency because it's already on chaoslib. Otherwise, I would have probably avoided it by default as it's heavy. But we should be fine.
> 
> Also, could you add an entry to the changelog?

Done, first time doing it. Is it ok?

File: chaostoolkit/cli.py [hunk 3]
```diff
@@ -385,6 +404,11 @@ def init(ctx: click.Context, discovery_path: str = "./discovery.json",
 with_plugins(iter_entry_points('chaostoolkit.cli_plugins'))(cli)
 
 
+def is_yaml(experiment_path: str):
```
Comment (Lawouach) [inline chaostoolkit/cli.py]: Missing the typing of the return value ;)

```
def is_yaml(experiment_path: str) -> bool:
```

File: tests/test_misc.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+from chaostoolkit.cli import is_yaml
```
Comment (Lawouach) [inline tests/test_misc.py]: Ordering usually is as follows:

```
stdlib

dependencies

local
```

so here:

```
import pytest

from chaostoolkit.cli import is_yaml
```

Comment (nstjelja) [reply tests/test_misc.py]: done

File: chaostoolkit/cli.py (hunk 3 above)
Comment (nstjelja) [reply chaostoolkit/cli.py]: done

File: chaostoolkit/cli.py [hunk 5]
```diff
@@ -3,6 +3,7 @@
 import decimal
 import io
 import json
+import yaml
```
Comment (Lawouach) [inline chaostoolkit/cli.py]: sorry to be a pain but this isn't a stdlib module so should go later in the import ordering.

**Pull request merged**